listeners.js

```diff
@@ -192,5 +192,29 @@ module.exports = function startListeners(ssb, onError) {
         }
       )
     )
+
+    // re-add missing people to a new epoch if the epoch creator didn't
+    // we're only doing this for the preferred epoch atm
+    pull(
+      ssb.db.query(
+        where(type('group/init')),
+        live({ old: true }),
+        toPullStream()
+      ),
+      pull.filter((msg) => isInitEpoch(msg)),
+      pull.drain(
+        (msg) => {
+          // TODO: add random timeout
+
+          // TODO: check if it's the preferred epoch. maybe do that live instead of the query?
+
+          console.log("TODO: re-add people if they're missing")
```

Comment:
what do we do if we fail with re-adding anyone at all? should we run this at an interval as well?

Reply:
should we run a `checkAndFixMemberTangleIntegrity()` at the start of every function that reads or writes to the member tangle?

Reply:
unresolved question but i'm thinking it'll be for another pr

Reply:
testing this here https://github.com/ssbc/ssb-tribes2/pull/126 and also added a task in the issue for running checks at the start of member functions